spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1648,6 +1663,8 @@ trait OptimisticTransactionImpl extends TransactionalWrite
       for (attemptNumber <- 0 to maxRetryAttempts) {
         try {
           val postCommitSnapshot = if (attemptNumber == 0) {
+            updatedCurrentTransactionInfo = InCommitTimestampUtils
+              .updateInCommitTimestampEnablementInfo(updatedCurrentTransactionInfo, commitVersion)
```

Comment:
I don't love how the logic to update ICT info is scattered in so many parts of the code. It would be really nice if we could find a simpler way.

For example -- can we move this inside `doCommit`? It has access to everything we need, and then we wouldn't need to worry about duplicating that code inside the conflict checker, because the retry path also calls `doCommit` after resolving conflicts. The commit info update is anyway far separated from this code, so we wouldn't give up any locality.

Downside is, `doCommit` doesn't currently transform the actions, so it's an unexpected place for such logic. But this spot is no better in that regard.

Reply:
It would be tempting to move the enablement check into `updateMetadataInternal`, where other table feature enablement checks already live, but it won't have the proposed commit timestamp yet.

Reply:
Moved this to one level up in the stack trace to `commitImpl`.